src/MusicTheory/ScaleClass.elm

```diff
@@ -0,0 +1,165 @@
+    , PentatonicScaleClassIntervals
+    , ScaleClass(..)
+    , aeolian
+    , diminishedHalfWhole
+    , diminishedWholeHalf
+    , ionian
+    , major
+    , majorPentatonic
+    , minor
+    , minorPentatonic
+    , wholeTone
+    )
+
+import MusicTheory.Interval as Interval exposing (Interval)
+
+
+type ScaleClass
+    = PentatonicScaleClass PentatonicScaleClassIntervals
+    | HexatonicScaleClass HexatonicScaleClassIntervals
+    | HeptatonicScaleClass HeptatonicScaleClassIntervals
+    | OctatonicScaleClass OctatonicScaleClassIntervals
+
+
+type alias PentatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    }
+
+
+type alias HexatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    , sixthDegree : Interval
+    }
+
+
+type alias HeptatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    , sixthDegree : Interval
+    , seventhDegree : Interval
+    }
+
+
+type alias OctatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    , sixthDegree : Interval
+    , seventhDegree : Interval
+    , eighthDegree : Interval
+    }
+
+
+major : ScaleClass
+major =
+    ionian
+
+
+minor : ScaleClass
+minor =
+    aeolian
+
+
+ionian : ScaleClass
+ionian =
+    HeptatonicScaleClass
+        { firstDegree = Interval.perfectUnison
+        , secondDegree = Interval.majorSecond
+        , thirdDegree = Interval.majorThird
+        , fourthDegree = Interval.perfectFourth
+        , fifthDegree = Interval.perfectFifth
+        , sixthDegree = Interval.majorSixth
+        , seventhDegree = Interval.majorSeventh
+        }
+
+
+aeolian : ScaleClass
+aeolian =
+    HeptatonicScaleClass
+        { firstDegree = Interval.perfectUnison
+        , secondDegree = Interval.majorSecond
+        , thirdDegree = Interval.minorThird
+        , fourthDegree = Interval.perfectFourth
+        , fifthDegree = Interval.perfectFifth
+        , sixthDegree = Interval.minorSixth
+        , seventhDegree = Interval.minorSeventh
+        }
+
+
+diminishedHalfWhole : ScaleClass
+diminishedHalfWhole =
+    OctatonicScaleClass
+        { firstDegree = Interval.perfectUnison
+        , secondDegree = Interval.minorSecond
+        , thirdDegree = Interval.augmentedSecond
+        , fourthDegree = Interval.majorThird
+        , fifthDegree = Interval.augmentedFourth
+        , sixthDegree = Interval.perfectFifth
+        , seventhDegree = Interval.majorSixth
+        , eighthDegree = Interval.minorSeventh
+        }
+
+
+diminishedWholeHalf : ScaleClass
+diminishedWholeHalf =
```

Comment:
I was able to find exact intervals from the Wikipedia sites in the PR description, so I think if we're able to continue finding intervals reliably, we should keep doing so.

If we end up having to define scales in terms of semitones instead, that could really introduce a lot of complication in how we represent scale steps, since, as you know, the pitch class N semitones away from the first pitch class has infinite possible spellings.

Reply:
Let's keep intervals then.